### ISSUE TITLE
 Coordinate not open in Google Maps

### ISSUE THREAD

**Issue by NorVoi** [penguin86/nextcloud-maps-client#17]
Hello, 

is there a possibility that I can't open the saved coordinates in Google Maps but in another app of my choice (e.g. OSM And or Organic Maps)?

NorVoi

**Comment by X-Raph-X** [comment]
I agree, it should be great to use this coordinates with Waze or osmand depending on the situation and the usage.

**Comment by thiswillbeyourgithub** [comment]
Unfortunately, the lack of this very feature is the reason I won't be relying on nextcloud maps for my very soon travel :/ is it possible to do a bounty ? also applies to #1

**Comment by ant0nwax** [comment]
Hello, could you please rephrase the Issue Main Title from "Coordinate not open in Google Maps"
to "Open Coordinates on touch with OsmAnd+"

Allow me to be your inspiration ;)

First, ensure OsmAnd is installed on the device. Then, use the following code snippet in your Android application to launch OsmAnd with a specific coordinate:

double latitude = 40.7128; // Example latitude
double longitude = -74.0060; // Example longitude

// Create a URI for OsmAnd with the coordinates
Uri uri = Uri.parse("geo:" + latitude + "," + longitude);

// Create an intent to open OsmAnd
Intent intent = new Intent(Intent.ACTION_VIEW, uri);
intent.setPackage("net.osmand"); // OsmAnd's package name

// Check if OsmAnd is available
if (intent.resolveActivity(getPackageManager()) != null) {
    startActivity(intent);
}

This code constructs a geo: URI with the specified latitude and longitude and sets the package to net.osmand, which is the official package name for OsmAnd.
 The resolveActivity() check ensures that OsmAnd is installed before attempting to launch it.

Alternatively, if you want to use OsmAnd for navigation from the current location to a specific coordinate, you can use a more specific intent that includes the q parameter for the destination:

String uriString = "geo:0,0?q=" + latitude + "," + longitude + "(Destination)";
Uri uri = Uri.parse(uriString);

Intent intent = new Intent(Intent.ACTION_VIEW, uri);
intent.setPackage("net.osmand");
if (intent.resolveActivity(getPackageManager()) != null) {
    startActivity(intent);
}

This method leverages the geo: URI scheme supported by OsmAnd, which allows it to interpret the coordinates and display them on the map.
 OsmAnd will then use its routing engine to provide navigation to the specified point.

Note that while this method opens OsmAnd with the coordinates, it does not set OsmAnd as the system-wide default navigation app.


AND HERE IS THE G×××LE MARKETING GENIUS:

Android does *NOT* yet provide a standard API to set a default navigation app for all intents; instead, users must manually choose the app each time or use a chooser intent that includes multiple navigation apps.
 However, by using the correct package name and URI scheme, you can reliably launch OsmAnd for navigation purposes.

**Comment by sim6** [comment]
Using an URL with geo scheme (e.g. geo:0,0) Android shows a menu with all applications that support the schema (like Maps, OsmAnd~, CoMaps...) and the user can select one to use always or only one time.

**Comment by sim6** [comment]
See also https://git.ichibi.eu/penguin86/nextcloud-maps-client/pulls/1
